serve/package.go

```diff
@@ -262,6 +264,64 @@ func (*PluginServe) copyDocs(distPath, docsPath string) error {
 	return nil
 }
 
+func (*PluginServe) validatePluginExports(pluginPath string) error {
+	s, err := os.Stat(pluginPath)
+	if err != nil {
+		return err
+	}
+	if !s.IsDir() {
+		return errors.New("plugin path must be a directory")
+	}
+
+	checkRelativeDirs := []string{"resources" + string(filepath.Separator) + "plugin", "plugin"}
+	foundDirs := []string{}
+	for _, dir := range checkRelativeDirs {
+		p := filepath.Join(pluginPath, dir)
+		s, err := os.Stat(p)
+		if err == nil && s.IsDir() {
+			foundDirs = append(foundDirs, dir)
+		}
+	}
+	if len(foundDirs) == 0 {
+		return fmt.Errorf("plugin directory must contain at least one of the following directories: %s", strings.Join(checkRelativeDirs, ", "))
+	}
+
+	foundVersion := false
+	for _, dir := range foundDirs {
+		p := filepath.Join(pluginPath, dir)
+		if err := filepath.WalkDir(p, func(path string, d os.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() || foundVersion {
+				return nil
+			}
+			if !strings.HasSuffix(strings.ToLower(d.Name()), ".go") {
+				return nil
+			}
+
+			f, err := os.Open(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			if ok, err := fileContains(f, "Version"); err != nil {
```

Comment:
could also have a regex that requires whitespace around `Version`...

Reply:
Or maybe a regex that checks for `Version\s*=`?

Reply:
Will add regex

Reply:
Regex in https://github.com/cloudquery/plugin-sdk/pull/1359/commits/cdb8b66c842e44d72d723fe6a0b79511dc745858

https://regex101.com/r/7Hm4PT/1